docs/static/persistent-queues.asciidoc

```diff
@@ -0,0 +1,127 @@
+[[persistent-queues]]
+=== Persistent Queues
+
+WARNING: This functionality is in beta and is subject to change. It should be deployed in production at your own risk.
+
+By default, Logstash uses in-memory bounded queues between pipeline stages
+(input → filter and filter → output) to buffer events. The size of these 
+in-memory queues is fixed and not configurable. If Logstash terminates unsafely,
+either as the result of a software failure or the user forcing an unsafe
+shutdown, it's possible to lose queued events. 
+
+To prevent event loss in these scenarios, you can configure Logstash to use
+persisent queues. With persistent queues enabled, Logstash persists buffered
+events to disk instead of storing them in memory. 
+
+Persistent queues are also useful for Logstash deployments that require high
+throughput and resiliency. Instead of deploying and managing a message
+broker, such as Redis, RabbitMQ, or Apache Kafka, to handle a mismatch in
+cadence between the shipping stage and the relatively expensive processing
+stage, you can enable persistent queues to buffer events on disk. The queue size
+is variable and configurable, which means that you have more control over
+managing situations that could result in back pressure to the source. 
+
+[[persistent-queues-advantages]]
+==== Advantages of Persistent Queues
+
+Using persistent queues:
+
+* Provides protection from losing in-flight messages when the Logstash process is shut down or the machine is restarted.
+* Handles the surge of events without having to use an external queueing mechanism like Redis or Kafka.
+* Provides an at-least-once message delivery guarantee. If Logstash is restarted
+while events are in-flight, Logstash will attempt to deliver messages stored
+in the persistent queue until delivery succeeds at least once. In other words,
+messages stored in the persistent queue may be duplicated, but not lost.
+
+[[persistent-queues-limitations]]
+==== Limitations of Persistent Queues
+
+The current implementation of persistent queues has the following limitations:
+
+* This version does not enable full end-to-end resiliency. Logstash only
```

Comment:
This version does not enable full end-to-end resiliency except for anything sent to beats input.